src/Grpc/JsonTranscoding/src/Microsoft.AspNetCore.Grpc.JsonTranscoding/GrpcJsonTranscodingOptions.cs

```diff
@@ -18,16 +19,19 @@ public sealed class GrpcJsonTranscodingOptions
     public GrpcJsonTranscodingOptions()
     {
         _unaryOptions = new Lazy<JsonSerializerOptions>(
-            () => JsonConverterHelper.CreateSerializerOptions(new JsonContext(JsonSettings, TypeRegistry)),
+            () => JsonConverterHelper.CreateSerializerOptions(new JsonContext(JsonSettings, TypeRegistry, DescriptorRegistry)),
             LazyThreadSafetyMode.ExecutionAndPublication);
         _serverStreamingOptions = new Lazy<JsonSerializerOptions>(
-            () => JsonConverterHelper.CreateSerializerOptions(new JsonContext(JsonSettings, TypeRegistry), isStreamingOptions: true),
+            () => JsonConverterHelper.CreateSerializerOptions(new JsonContext(JsonSettings, TypeRegistry, DescriptorRegistry), isStreamingOptions: true),
             LazyThreadSafetyMode.ExecutionAndPublication);
     }
 
     internal JsonSerializerOptions UnarySerializerOptions => _unaryOptions.Value;
     internal JsonSerializerOptions ServerStreamingSerializerOptions => _serverStreamingOptions.Value;
 
+    // Registry is set by DI during startup.
+    internal DescriptorRegistry DescriptorRegistry { get; set; } = default!;
```

Comment:
Why does this need to be set in options?

Reply:
So it can be used by `JsonConverterHelper.CreateSerializerOptions` in the callbacks registered by this type's constructor.